Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -2,35 +2,1345 @@
+   (NonAssocSemiring.toAddCommMonoidWithOne.{u1} R _inst_1)))) n a)
+    (HMul.hMul.{u1, u1, u1} R R R (instHMul.{u1} R (Distrib.toHasMul.{u1} R
+    (NonUnitalNonAssocSemiring.toDistrib.{u1} R
+    (NonAssocSemiring.toNonUnitalNonAssocSemiring.{u1} R _inst_1))))
+     ((fun (a : Type)
+    (b : Type.{u1}) [self : HasLiftT.{1, succ u1} a b] => self.0) Nat R
+    (HasLiftT.mk.{1, succ u1} Nat R (CoeTCₓ.coe.{1, succ u1} Nat R
+    (Nat.castCoe.{u1} R (AddMonoidWithOne.toNatCast.{u1} R
+    (AddCommMonoidWithOne.toAddMonoidWithOne.{u1} R
+    (NonAssocSemiring.toAddCommMonoidWithOne.{u1} R _inst_1)))))) n) a)
+but is expected to have type
+  forall {R : Type.{u1}} [_inst_1 : Semiring.{u1} R] (n : Nat) (a : R),
+   Eq.{succ u1} R (HSMul.hSMul.{0, u1, u1} Nat R R (instHSMul.{0, u1}
+    Nat R (AddMonoid.SMul.{u1} R (AddMonoidWithOne.toAddMonoid.{u1} R
+    (AddCommMonoidWithOne.toAddMonoidWithOne.{u1} R (NonAssocSemiring.toAddCommMonoidWithOne.{u1} R
+    (Semiring.toNonAssocSemiring.{u1} R _inst_1)))))) n a) (HMul.hMul.{u1, u1, u1}
+     R R R (instHMul.{u1} R (NonUnitalNonAssocSemiring.toMul.{u1} R
+     (NonAssocSemiring.toNonUnitalNonAssocSemiring.{u1} R
+     (Semiring.toNonAssocSemiring.{u1} R _inst_1))))
+     (Nat.cast.{u1} R (Semiring.toNatCast.{u1} R _inst_1) n) a)
+Case conversion may be inaccurate. Consider using '#align nsmul_eq_mul nsmul_eq_mulₓ'. -/
+@[simp]
+theorem nsmul_eq_mul [NonAssocSemiring R] (n : ℕ) (a : R) : n • a = n * a := by
   rw [nsmul_eq_mul', (n.cast_commute a).eq]
+#align nsmul_eq_mul nsmul_eq_mul
+
+/-- Note that `add_comm_monoid.nat_smul_comm_class` requires stronger assumptions on `R`. -/
+instance NonUnitalNonAssocSemiring.nat_smul_comm_class [NonUnitalNonAssocSemiring R] :
+    SMulCommClass ℕ R R :=
+  ⟨fun n x y => by
+    induction' n with n ih
+    · simp [zero_nsmul]
+    · simp_rw [succ_nsmul, smul_eq_mul, mul_add, ← smul_eq_mul, ih]⟩
+#align
+  non_unital_non_assoc_semiring.nat_smul_comm_class NonUnitalNonAssocSemiring.nat_smul_comm_class
+
+/-- Note that `add_comm_monoid.nat_is_scalar_tower` requires stronger assumptions on `R`. -/
+instance NonUnitalNonAssocSemiring.nat_is_scalar_tower [NonUnitalNonAssocSemiring R] :
+    IsScalarTower ℕ R R :=
+  ⟨fun n x y => by
+    induction' n with n ih
+    · simp [zero_nsmul]
+    · simp_rw [succ_nsmul, ← ih, smul_eq_mul, add_mul]⟩
+#align
+  non_unital_non_assoc_semiring.nat_is_scalar_tower NonUnitalNonAssocSemiring.nat_is_scalar_tower
+
+@[simp, norm_cast]
+theorem Nat.cast_pow [Semiring R] (n m : ℕ) : (↑(n ^ m) : R) = (↑n : R) ^ m := by
+  induction' m with m ih
+  · simp
+  · rw [_root_.pow_succ', _root_.pow_succ', Nat.cast_mul, ih]
+#align nat.cast_pow Nat.cast_pow
+
+-- Porting note: `norm_cast` attribute removed.
+/- Porting note `simp` attribute removed as suggested by linter:
+simp can prove this:
+  by simp only [Nat.cast_pow]
+-/
+-- Porting note: commented out as linter reports LHS= RHS syntactically
+-- -- @[simp]
```

Comment:
These porting notes seem to conflict / overlap?

Reply:
Came up in different rounds of linting. Will try to fix.